v2realbot/utils/utils.py

```diff
@@ -62,21 +62,13 @@ def validate_and_format_time(time_string):
 
 def fetch_calendar_data(start, end):
     nyse = mcal.get_calendar('NYSE')
-    cal_dates = []
     schedule = nyse.schedule(start_date=start, end_date=end, tz='America/New_York')
-    schedule_dict = schedule.reset_index().rename(columns={"index": "date", 
-                                                           "market_open": "open",
-                                                           "market_close": "close"}).to_dict(orient="records")
-    print(schedule_dict)
-
-    for day in schedule_dict:
-        date = pd.to_datetime(day["date"]).replace(tzinfo=None).date()
-        open = pd.to_datetime(day["open"]).replace(tzinfo=None)
-        close = pd.to_datetime(day["close"]).replace(tzinfo=None)
-
-        calendar = Calendar(date=date, open=open, close=close)
-        cal_dates.append(calendar)
-    print(cal_dates)
+    schedule = (schedule.reset_index()
+                        .rename(columns={"index": "date", "market_open": "open", "market_close": "close"})
+                        .assign(open=lambda day: pd.to_datetime(day['open']).dt.tz_localize(None),
```

Comment:
jelikož ten objekt už je datetime, myslím, že pd.to_datetime funkce jen tam redundantní a stačila by ta lokalizace.

Reply:
Neni mozne ze tato chyba se objevuje pokud je dataframe prazdny v pripade, kdy neni obchodni den stejne tak jako dnes ? Mozna proto by asi bylo lepsi ponechat pd.to_datetime() pred samotnou lokalizaci ...

Reply:
Jj to je možný, zkus ověřit, jestli se transformace neprovádí i pro prázdný dataframe. To by stačilo jen ošetřit.

Reply:
Osetril jsem to, ze pokud je dataframe prazdny, funkce rovnout vrati prazdny list jako cal_dates. Zde ale narazim na chybu, ktera se tyka rozbaleni slovniku pri definici Calendar objektu - Calendar(**record) . Calendar objekt ma v sobe __init__ methodu, ktera ze stringu vytvari datetime objekty .... to je asi ta zabudovana tranformace do datetime, o ktere jsme mluvili: 
def __init__(self, **data: Any) -> None:
        if "date" in data and "open" in data:
            start_datetime_str = data["date"] + " " + data["open"]
            data["open"] = datetime.strptime(start_datetime_str, "%Y-%m-%d %H:%M")

        if "date" in data and "close" in data:
            start_datetime_str = data["date"] + " " + data["close"]
            data["close"] = datetime.strptime(start_datetime_str, "%Y-%m-%d %H:%M")

Ma s tim problem, protoze nedokaze pouzit scitani + pro objekty Timestamp. Alpaca API vraci pravdepodobne JSON string, ktery je preveden do datetime...